bridges/centralized-ethereum/src/actors/dr_sender.rs

```diff
@@ -66,42 +65,84 @@ impl DrSender {
     /// Initialize the `DrSender` taking the configuration from a `Config` structure
     /// and a Json-RPC client connected to a Witnet node
     pub fn from_config(config: &Config, node_client: Addr<JsonRpcClient>) -> Self {
-        let max_dr_value_nanowits = config.max_dr_value_nanowits;
-        let wit_dr_sender_polling_rate_ms = config.wit_dr_sender_polling_rate_ms;
-        let dr_fee_nanowits = config.dr_fee_nanowits;
-
         Self {
+            polling_rate_ms: config.eth_new_drs_polling_rate_ms / 2 + 1000,
             witnet_client: Some(node_client),
-            wit_dr_sender_polling_rate_ms,
-            max_dr_value_nanowits,
-            dr_fee_nanowits,
+            witnet_dr_min_collateral_nanowits: config.witnet_dr_min_collateral_nanowits,
+            witnet_dr_max_value_nanowits: config.witnet_dr_max_value_nanowits,
+            witnet_dr_max_fee_nanowits: config.witnet_dr_max_fee_nanowits,
+            witnet_node_pkh: None,
         }
     }
 
     fn check_new_drs(&self, ctx: &mut Context<Self>, period: Duration) {
         let witnet_client = self.witnet_client.clone().unwrap();
-        let max_dr_value_nanowits = self.max_dr_value_nanowits;
-        let dr_fee_nanowits = self.dr_fee_nanowits;
+        let witnet_dr_min_collateral_nanowits = self.witnet_dr_min_collateral_nanowits;
+        let witnet_dr_max_value_nanowits = self.witnet_dr_max_value_nanowits;
+        let witnet_dr_max_fee_nanowits = self.witnet_dr_max_fee_nanowits;
+        let mut witnet_node_pkh = self.witnet_node_pkh.clone();
 
         let fut = async move {
             let dr_database_addr = DrDatabase::from_registry();
             let dr_reporter_addr = DrReporter::from_registry();
 
+            if witnet_node_pkh.is_none() {
+                // get witnet node's pkh if not yet known
+                let req = jsonrpc::Request::method("getPkh").timeout(Duration::from_millis(5000));
+                let res = witnet_client.send(req).await;
+                witnet_node_pkh = match res {
+                    Ok(Ok(res)) => match serde_json::from_value::<String>(res) {
+                        Ok(pkh) => {
+                            log::info!("Pkh is {}", pkh);
+
+                            Some(pkh)
+                        }
+                        Err(_) => None,
+                    },
+                    Ok(Err(_)) => {
+                        log::warn!("Cannot deserialize witnet node's pkh, will retry later");
+
+                        None
+                    }
+                    Err(_) => {
+                        log::warn!("Cannot get witnet node's pkh, will retry later");
+
+                        None
+                    }
+                };
+            } else {
+                // TODO: alert if number of big enough utxos is less number of drs to broadcast
```

Comment:
What's missing to uncomment this?

Reply:
The implementation of this feature requires deserializing JSON object returned by `getUtxoInfo`. It should be tackled on a different issue and PR.